joomla/joomla-cms
# Joomla 3.10 EOS Plugin

Pull Request for Issue #31842

### Summary of Changes

Based on the proposal by @softforge and together with the @joomla/cms-release-team here is a proposal for the 3.10 EOS Plugin to be shipped with 3.10.0.

With the stable release of Joomla 3.10.0 there will be a fixed EOS date (release date + 2 years). This plugin aims to notify the admins first a bit discrete as additional quickicon that point to informations about Joomla 4 and later as notice that 3.10 once it enters security only mode and that at the end of support is "arround the dor" and once its reached as message above the cpanel.

PS. Yes there is a date in the plugin now but thats not the final one and will be changed once the final release is out.

### Testing Instructions

#### Test messages

To be updated once the updated texts are provided

#### Test the upgrade

- Install 3.10.0-alpha6: https://github.com/joomla/joomla-cms/releases/tag/3.10.0-alpha6
- Install this package using upload & install within com_joomlaupdate: https://ci.joomla.org/artifacts/joomla/joomla-cms/3.10-dev/34453/downloads/45034/Joomla_3.10.0-alpha7-dev+pr.34453-Development-Update_Package.zip
- Notice that there is a new plugin called `Quick Icon - Joomla! 3.10 End Of Support Notification` (Extensions -> Plugins -> Filter type: "quickicon")
![image](https://user-images.githubusercontent.com/2596554/121067900-e5a4bf80-c7cb-11eb-84e7-4c5785c96173.png)

### Actual result BEFORE applying this Pull Request

No EOS plugin shipped with 3.10

### Expected result AFTER applying this Pull Request

There is a EOS plugin shipped with 3.10

### Documentation Changes Required

The EOS plugin and its messages needs to be documented. There is also an magazin article planed once this here is finalized.

### Questions

@brianteeman can you please review at the proposed language strings.
@richard67 please check the SQL stuff hopefully i have not messed something up?

### Once this PR here is ready

- [ ] ... there is a magazin article planed to be released with the (than) next magazin issue.
- [ ] ... there will be a PR against 4.0-dev to make sure this plugin is uninstalled once upgraded to Joomla 4 similiar to the 2.5 plugin.

### Thanks

A big thanks goes to @softforge and the @joomla/cms-release-team where we discussed some proposals and where this plugin here is now based on.
